### ISSUE TITLE
[FR] cc_libraries should use "includes", not "strip_include_prefix"

### ISSUE THREAD

**Issue by carlosgalvezp** [google/benchmark#1512]
So that when users import google-benchmark, the headers are included as "system headers", instead of regular headers.

If this is not done, compilers will throw warnings at google-benchmark code, which users have no way to act upon. Users should be able to choose a highly restrictive set of warnings without having to fix them in google-benchmark.

Note that this pattern is already applied in the googletest repo.

**Comment by varshneydevansh** [comment]
Is the below discussion related to the above issue?

[Bazel "strip_include_prefix" or "includes" attributes in cc_import rules](https://github.com/bazelbuild/bazel/issues/4748) suggested solution in this thread -

1. https://github.com/bazelbuild/bazel/issues/4748#issuecomment-515887921
2. https://github.com/bazelbuild/bazel/issues/4748#issuecomment-1260781555

 
I would like to work on this after completing my pending PR.


**I went on reading what strip_include_prefix is?**
- Functionality of the "strip_include_prefix" function provides a way to remove a specific prefix from a string.

**Comment by carlosgalvezp** [comment]
Hi! No, those links are related to cc_import, while I'm referring to cc_library in this ticket :)

**Comment by varshneydevansh** [comment]
Yes that bazel issue is about `cc_import` but the workaround for that and with the `cc_library` they used `includes` instead of `strip_include_prefix`. So, I found it a good starting point.

I am looking more into this. https://groups.google.com/g/bazel-discuss/c/pQLmBVNJGZ4?pli=1
